Sandbox/src/Game/Definitions/NpcShopDefinition.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "ItemStock.h"
+
+class NpcShopDefinition
+{
+private:
+	int mIndex;
+	std::string mName;
+	bool mBuysAll;
+	std::vector<ItemStock> mStock;
+
+	NpcShopDefinition(int index, std::string name, bool buysAll, std::vector<ItemStock> stock)
+	{
+		mIndex = index;
+		mName = name;
+		mBuysAll = buysAll;
+		mStock = stock;
+	}
+
+	~NpcShopDefinition() = default;
+
+	friend class NpcRepository;
+
+public:
+	inline int Index() const { return mIndex; }
+	inline std::string Name() const { return mName; }
+	inline bool BuysAll() const { return mBuysAll; }
```

Comment:
what's this for?

Reply:
Do you mean the class itself? If an Npc sell items, it will store the index of that shop. This will let the game know what items it has to sell. The BuysAll value determines if they buy all items or just items to replenish their stock.